### DISCUSSION TITLE
Control which button of a Button Matrix is toggled?

### DISCUSSION THREAD

**Discussion by ronkmd** [HASwitchPlate/openHASP#180] [Q&A]
Is there a way to control which button of a Button Matrix is toggled? I'd like to read current fan speed (off,lo,med,high) attribute and toggle the appropriate button in the Button Matrix. I can read the button presses (val=0,1,2 or3) but I want to set the initial state of the Button Matrix according to the fan speed attribute. Ideas?

**Comment by nagyrobi** [comment]
This is fixed in the master branch, but not yet released. You can try it though with a nightly build from the Actions tab.

One thing we need to add, is to have a way to un-toggle all the buttons when, for example your fan in off. This is still under thinking.

**Reply by ronkmd** [reply]
Thanks. I'll probably wait for the release.

**Reply by nagyrobi** [reply]
In the main branch (pre 0.6.1) this is fixed:
- you can set with val which buttons to appear pressed
- you can use `val=-1` to un-toggle the entire button matrix (no button pressed) (useful for stopped/disabled/unavailable state of your fan, for example)

**Reply by ronkmd** [reply]
Thanks!
